src/pipx/shared_libs.py

```diff
@@ -30,6 +31,15 @@ def __init__(self) -> None:
         # i.e. python_path is ~/.local/pipx/shared/python
         self._site_packages: Optional[Path] = None
         self.has_been_updated_this_run = False
+        # TODO: remove setuptools (wheel?)
+        self.required_packages = [
+            "pip",
+            "wheel",
+            "packaging",
+            "importlib-metadata",
+            "setuptools",
```

Comment:
Why do we need packaging and importlib metadata as shared lib?

Reply:
This is the major change to `venv_metadata_inspector.py`.  Instead of it using setuptools' `pkg_resources` to inspect each venv, it now uses `packaging` and `importlib-metadata`.

From `packaging.requirements` it uses `Requirement`, and from `packaging.utils` it uses `canonicalize_name()`.
From `importlib[.-]metadata` it uses `distribution()`, `distributions()`, and type `Distribution`. 

I couldn't figure out a way to get along without both of those dependencies if we transition away from `pkg_resources`.  If anyone can think of a way I'd be happy to pursue it!  Adding more libraries to the shared libs does increase the chance that we will conflict with requirements in the venvs.

Reply:
I think we can avoid `importlib-metadata` in 3.8+ environments by adding a marker here (plus try-except imports in `venv_metadata_inspector`)?

Reply:
I was guarding against the user possibly having a mix of 3.8+ and <3.8 venvs, in which case the shared libs would need `importlib-metadata` even if pipx and possibly other venvs were running in 3.8+.

Reply:
Although now that I think about it, if the shared libs are installed 3.8+ but used with a venv that is <3.8, does that imply that the <3.8 venvs get to use `importlib.metadata`?

Reply:
And there is already a try-except sequence in `venv_metadata_inspector.py` to first attempt to use `importlib.metadata` and then to fall back on using `importlib-metadata`